src/pages/index.js

```diff
@@ -3,15 +3,82 @@ import FormValidator from "../components/FormValidator.js";
 import PopupWithImage from "../components/PopupWithPictures.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import UserInfo from "../components/UserInfo.js";
+import Api from "../components/Api.js";
 import Section from "../components/Section.js";
+import PopupWithConfirm from "../components/PopupWithConfirm.js";
 import { initialCards, config } from "../utils/constants.js";
 import "../pages/index.css"
 
-//---------------------RENDERING INITIAL CARDS-------------------
 
+// Select DOM elements
+const profileEditBtn = document.getElementById("profile__edit-button");
+const profileNameInput = document.querySelector("[name='title']");
+const profileDescInput = document.querySelector("[name='about']");
+const addModalOpenBtn = document.querySelector(".profile__add-button");
+
+const pictureNameInput = document.querySelector("[name='name']");
+const pictureUrlInput = document.querySelector("[name='url']")
+const pictureSubmitButton = document.querySelector("#picture_submit");
+
+const avatarEditBtn = document.querySelector(".profile__avatar-button");
+
+const deleteButton = document.querySelectorAll(".card__delete");
+const confirmSubmitBtn = document.querySelector("#confirmation__modal_submit");
+const confirmCloseBtn = document.querySelector("#confirmation__modal_close");
+
+const likeButtons = document.querySelectorAll(".card__like-button");
+
+const profileName = document.querySelector('.profile__title');
+const profileDesc = document.querySelector('.profile__description');
+const profileAvatar = document.querySelector('.profile__avatar');
+
+
+//---------------------API GOODS------------------------------------
+
+// Select all submit buttons with the class "modal__submit"
+const submitButtons = document.querySelectorAll('.modal__submit');
+
+// Add an event listener to each button
+submitButtons.forEach((button) => {
+  button.addEventListener('click', function() {
+    
+    const originalText = this.textContent;
+    
+    
+    this.textContent = 'saving';
+    
+    // After 2 seconds, revert the text back to the original. I sure hope this 
+    // is what you meant 
+    setTimeout(() => {
+      this.textContent = originalText;
```

Comment:
Please, delete it

You need to **only** return the **default** button text in the `finally` block, so that the button could get the text regardless of whether the request succeeds or fails.